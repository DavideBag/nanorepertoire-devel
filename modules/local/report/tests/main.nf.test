nextflow_process {

    name "Test Process REPORT"
    script "../main.nf"
    process "REPORT"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "report"

    test("reporting") {

        when {
            process {
                """
                // input_ch
                input[0] = [
                    file("${moduleDir}/analysis_report.qmd")
                ]
                input[1] = [
                    file("${moduleDir}/loop_tree.qmd")
                ]
                input[2] = [
                    file('/workspace/nanorepertoire/.nf-test/tests/9b7bcb08d25bcc630df61136bfa4169a/work/aa/ff026bc963fce799c9f64e8e0a2f4e/test_clusters.summary', checkIfExists: true)
                ]
                input[3] = [
                    file('/workspace/nanorepertoire/.nf-test/tests/9b7bcb08d25bcc630df61136bfa4169a/work/be/84da1e9b14388fd23623eebee47fec/test_cdr3.hist', checkIfExists: true)
                ]
                input[4] = [
                    file('/workspace/nanorepertoire/.nf-test/tests/9b7bcb08d25bcc630df61136bfa4169a/work/be/84da1e9b14388fd23623eebee47fec/test_cdr3.tsv', checkIfExists: true)
                ]
                input[5] = [
                    [id: 'test', sampleID: 'sample1', individualID: 'ind1', immunisation: 'imm1', boost: 'boost1']                ]
                input[6] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.stdout).match()}
            )
        }
    }
}
