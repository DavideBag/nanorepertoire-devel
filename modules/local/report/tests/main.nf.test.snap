{
    "reporting": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-18T10:59:56.636871"
    }
}